ibm/resource_ibm_cbr_zone.go

```diff
@@ -0,0 +1,682 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/contextbasedrestrictionsv1"
+)
+
+func resourceIBMCbrZone() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMCbrZoneCreate,
+		ReadContext:   resourceIBMCbrZoneRead,
+		UpdateContext: resourceIBMCbrZoneUpdate,
+		DeleteContext: resourceIBMCbrZoneDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
```

Comment:
there is no required field in this resource
this resource crashes when no arguments are provided in the resource block